airflow/jobs/scheduler_job.py

```diff
@@ -521,14 +521,24 @@ def _process_executor_events(self, session: Session = None) -> int:
                 )
                 self.log.error(msg, ti, state, ti.state, info)
 
-                request = TaskCallbackRequest(
-                    full_filepath=ti.dag_model.fileloc,
-                    simple_task_instance=SimpleTaskInstance(ti),
-                    msg=msg % (ti, state, ti.state, info),
-                )
-                self.log.info('Setting task instance %s state to %s as reported by executor', ti, state)
-                ti.set_state(state)
-                self.processor_agent.send_callback_to_execute(request)
+                # Get task from the Serialized DAG
+                try:
+                    dag = self.dagbag.get_dag(ti.dag_id)
+                    task = dag.get_task(ti.task_id)
+                except Exception as ex:
+                    self.log.exception("Marking task instance %s as failed. Reason: %s", ti, ex)
```

Comment:
`log.exception` already includes the exception, so we don't need to include the message again.
```suggestion
                    self.log.exception("Marking task instance %s as %s", ti, state)
```